tests/robustness/multiclient_test/framework/harness.go

```diff
@@ -296,3 +297,51 @@ func (th *TestHarness) Cleanup(ctx context.Context) (retErr error) {
 
 	return retErr
 }
+
+// GetDirsToLog collects the directory paths to log.
+func (th *TestHarness) GetDirsToLog(ctx context.Context) []string {
+	if th.snapshotter == nil {
+		return nil
+	}
+
+	var dirList []string
+	dirList = append(dirList,
+		th.dataRepoPath, // repo under test base dir
+		th.metaRepoPath, // metadata repository base dir
+		path.Join(th.fileWriter.DataDirectory(ctx), ".."), // LocalFioDataPathEnvKey
+		th.engine.MetaStore.GetPersistDir(),               // kopia-persistence-root-
+		th.baseDirPath,                                    // engine-data dir
+	)
+
+	cacheDir, _, err := th.snapshotter.GetCacheDirInfo()
+	if err == nil {
+		cacheDir = strings.Trim(cacheDir, "\n")
+		dirList = append(dirList, cacheDir) // cache dir for repo under test
+	}
+	allCacheDirs := getAllCacheDirs(cacheDir)
+	dirList = append(dirList, allCacheDirs...)
+
+	log.Printf("in GetDirsToLog")
+	for _, d := range dirList {
+		log.Printf("dir %s", d)
+	}
+
+	return dirList
+}
+
+func getAllCacheDirs(dir string) []string {
+	if dir == "" {
+		return nil
+	}
+	var dirs []string
+	// Collect all cache dirs
+	// There are six types of caches, and corresponding dirs.
+	// metadata, contents, indexes,
+	// own-writes, blob-list, server-contents
+	cacheDirSubpaths := []string{"metadata", "contents", "indexes", "own-writes", "server-contents"}
```

Comment:
nit; you might be able to create a constant shared between `cli/command_cache_clear.go` and this file.

Reply:
I will address this in a follow up PR.